### DISCUSSION TITLE
[8.0 QA] Essentials

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25687] [Announcement]
#### Feature owner: @ndelangen 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by valentinpalkovic** [comment]
Create a Story file, paste valid content into it and then delete it again. Independently of whether you are navigating to another Story, Storybook doesn't recover from this error state:

![image](https://github.com/storybookjs/storybook/assets/5889929/7d9d01b5-c0e5-432e-ad8b-30deb7b820dc)

**Reply by vanessayuenn** [reply]
sev: s3

**Comment by shilman** [comment]
**Project** Storyblok
**Version** 8.0.0-alpha.13

`Source` block renders with `source.type = 'source'` for `SbButton` component, even though the story accepts args. Manually setting `source.type = 'dynamic'` results in no source appearing

<img width="1119" alt="image" src="https://github.com/storybookjs/storybook/assets/488689/4070764f-7b3e-40fd-a34f-9c894779f09a">

**Reply by vanessayuenn** [reply]
sev: S2, but not beta-blocking

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

HMR on docs when changing the Meta title results in extra entries on the sidebar:

![2024-01-25 12 41 45](https://github.com/storybookjs/storybook/assets/1671563/6c2a4a62-a26c-40b2-89f1-70a96d0ce299)

**Repro:**
1. Go to https://github.com/yannbf/mealdrop/tree/storybook/8.0.0
2. Open Colors.mdx and its story
3. Change the title like in the video

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

HMR on docs makes the backgrounds icon disappear sometimes:

![2024-01-25 12 40 55](https://github.com/storybookjs/storybook/assets/1671563/5d8c1110-9293-4779-a592-6b481e16f072)

In general, I guess docs only MDX should never have any icons in the toolbar, or at least essential addon icons. Perhaps icons via custom toolbar should have an option to show/hide in this scenario.

**Repro:**
1. Go to https://github.com/yannbf/mealdrop/tree/storybook/8.0.0
2. Open any mdx story and its file
3. Trigger HMR

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

When renaming the story file of which you are visiting, a lot of weird things happen, it's best described in a video:

https://github.com/storybookjs/storybook/assets/1671563/540e03b8-82bf-4893-a0e3-adfe8058cf3f

The issues which happen are:
1 - The story of which you renamed somehow changes order and goes to the bottom of the list (there is no storySort being used)
2 - An `importers[path] is not a function` error is flashed in the screen:
<img width="349" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/5c5dc9c3-40c1-4252-a6fb-da45a5f9768f">

3 - The UI goes into a funky state where there's some sort of negative margin and the toolbar is unreachable (until you close the addon panel, which is not really closed)

**Repro:**
1. Go to https://github.com/yannbf/mealdrop/tree/storybook/8.0.0
2. Visit a story
3. Rename its story file to something else (e.g. add an A at the end)

**Reply by vanessayuenn** [reply]
tracked in #25788

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

When renaming an MDX story file (even if you are not visiting that story), Storybook gets into a half crashing state, best described in a video:


https://github.com/storybookjs/storybook/assets/1671563/7c146dda-0694-43c9-8f6e-19daa567508c

It seems that there are some issues in event emitting and error handling, related to fetching index.json, the sidebar, and HMR from docs files.

**Repro:**
1. Go to https://github.com/yannbf/mealdrop/tree/storybook/8.0.0
2. Visit the badge story
3. Rename `Button.stories.ts` to `Buttons.stories.ts`

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

Number controls do not reset:

<img src="https://github.com/storybookjs/storybook/assets/1671563/924f1fd0-af6e-47bc-ace9-d86301d564c5" width="300"/>

**Repro:**
1. Go to https://stackblitz.com/edit/github-cde8qu?file=src%2Fstories%2FButton.jsx,src%2Fstories%2FButton.stories.js&preset=node
2. Visit a Button story
3. Define value for the `age` control
4. Reset it
5. Same issue happens in ArgsTable in autodocs

**Reply by vanessayuenn** [reply]
tracked by https://github.com/storybookjs/storybook/issues/25819
